microcosm_flask/conventions/relation.py

```diff
@@ -41,11 +42,56 @@ def configure_createfor(self, ns, definition):
         @response(definition.response_schema)
         def create(**path_data):
             request_data = load_request_data(definition.request_schema)
-            response_data = definition.func(**merge_data(path_data, request_data))
+            response_data = require_response_data(definition.func(**merge_data(path_data, request_data)))
             return dump_response_data(definition.response_schema, response_data, Operation.CreateFor.value.default_code)
 
         create.__doc__ = "Create a new {} relative to a {}".format(pluralize(ns.object_name), ns.subject_name)
 
+    def configure_deletefor(self, ns, definition):
+        """
+        Register a delete-for relation endpoint.
+
+        The definition's func should be a delete function, which must:
+        - accept kwargs for path data
+        - return truthy/falsey
+
+        :param ns: the namespace
+        :param definition: the endpoint definition
+
+        """
+        @self.graph.route(ns.instance_path, Operation.DeleteFor, ns)
+        def delete(**path_data):
+            require_response_data(definition.func(**path_data))
+            return "", Operation.DeleteFor.value.default_code
+
+        delete.__doc__ = "Delete a {} relative to a {}".format(pluralize(ns.object_name), ns.subject_name)
+
+    def configure_replacefor(self, ns, definition):
```

Comment:
@jessemyers you mean, once we have a concrete object id we can define relevant routes directly on it for things like Replace? 
That makes sense generally, however specifically for DynamoDB we ran into a situation where the DynamoDB API requires the composite key of (crawl_job_id, result_id) to access a 'result' (here crawl_job would be the subject, result would be the object of the relation). 
so the most straight-forward way of doing that seems to be to make it part of the URL scheme as in /api/crawl_job/:crawl_job_id/result/:id  - The alternative could be to require passing 'crawl_job_id' as a body field in request for /api/result/:result_id I suppose, but that seems much uglier. WDYT?